src/dataflow/src/render/reduce.rs

```diff
@@ -736,11 +736,22 @@ where
     G: Scope,
     G::Timestamp: Lattice,
 {
-    collection.reduce_abelian::<_, OrdValSpine<_, _, _, _>>("DistinctBy", {
-        |key, _input, output| {
-            output.push((key.clone(), 1));
+    let negated_result = collection.reduce_named("DistinctBy", {
+        |key, input, output| {
+            output.push((key.clone(), -1));
+            output.extend(
+                input
+                    .iter()
+                    .map(|(values, count)| ((*values).clone(), *count)),
+            );
```

Comment:
Why do we have to output things that are on our input, in addition to the key?

Reply:
I'm not certain I understand the question (afaict, it is about code you wrote, right; I might be misunderstanding). But, what we are about to do is subtract this output from the input, and we want to subtract everything except the `key`. So, a negative key produced as output here "unsubtracts" it.

I think for this operator things could probably be simpler, in that we can probably assert that `input` has length exactly one, `count` is strictly greater than zero, and what we want to produce as output is `key` with multiplicity `count - 1`.

Reply:
why would we be able to assert that input has length `1` -- I looked around in the file and couldn't see where that was enforced maybe its enforced somewhere else. Otherwise I agree with everything else

Reply:
Answering my own question: In order to only produce `(key.clone(), -1)`, we need to subtract the original inputs from our outputs. Thanks Frank for pointing me in the right direction!